### ISSUE TITLE
Feature request: moratorium on cipher deprecations

### ISSUE THREAD

**Issue by moderation** [projectcontour/contour#3299]
Contour 1.11.0 deprecated some older TLS ciphers and that change didn't make it into the release notes - https://github.com/projectcontour/contour/releases/tag/v1.11.0. The specific PR is https://github.com/projectcontour/contour/pull/3154 which removed `ECDHE-RSA-AES128-SHA` and `ECDHE-RSA-AES256-SHA` ciphers.

In our enterprise environment we have Contour configured to only allow >= TLS 1.2. However the cipher removal broke:
- IE11 on Windows 7 virtual desktops ability to negotiate TLS 1.2 with Contour
- Windows Server 2012 R2 .NET or IE11 clients ability to negotiate TLS 1.2 with Contour

Request:
- ensure any breaking changes regarding TLS ciphers, version etc. make the release notes
- revert https://github.com/projectcontour/contour/pull/3237 from release 1.12.0 until a full investigation of the enterprise platforms that the PR may effect
- introduce a moratorium on cipher deprecation's until https://github.com/projectcontour/contour/issues/2880 lands

/cc @youngnick as per Slack chat

**Comment by youngnick** [comment]
Thanks for this @moderation. In this case, I agree we need to roll back #3237 for 1.12.0, while we work on finishing out #2880. @sunjayBhatia already has #3292 which gets started on that.

I'd like some input from the other maintainers here though? @stevesloka @skriss @xaleeks @danehans ?

**Comment by tsaarni** [comment]
> revert #3237 from release 1.12.0 until a full investigation of the enterprise platforms that the PR may effect

Quick comment about the effects:

#3237 impacts the same platforms as #3154 since these ciphers should go together as a pair.  

The difference between the two sets of ciphers is based on which type of keys are used in the server certificate: if server certificate with RSA key is used, that will allow the server to negotiate RSA variants of the ciphers, and server certificate with EC key will allow server to negotiate the EC variants of the ciphers:

| cipher with RSA server certificate | cipher with EC server certificate |
| - | - |
| ECDHE-RSA-AES128-SHA | ECDHE-ECDSA-AES128-SHA |
| ECDHE-RSA-AES256-SHA | ECDHE-ECDSA-AES256-SHA |

**Comment by youngnick** [comment]
We reverted both PRs as part of the 1.12 release, we will bring them back in once #3292 is done, and look towards finishing up #2880 completely in the 1.13 timeframe.

**Comment by sunjayBhatia** [comment]
How do we feel about being FIPS etc. compliant by default and this sort of migration path?
- release 1.13.0 with configurable ciphers which should give users the ability to start using the feature, and configure exactly the ciphers they need
- release 1.14.0 with https://github.com/projectcontour/contour/pull/3304 reapplied (and possibly more removals to be FIPS compliant by default)

**Comment by youngnick** [comment]
That that FIPS compliance by default is actually not as secure as our previous defaults, because FIPS lags quite a bit behind state-of-the-art.

I think that we are better off bringing the previous default removals back in, as long as we document what we are changing, and provide information on how to put the ciphersuite back to how it is in 1.12.

To put this another way, I don't think we should wait to fix the defaults once we have configurable ciphers available, we just need to ensure that the docs and changelog correctly note the changes (and provide the exact ciphers for people to add back in).

**Comment by sunjayBhatia** [comment]
See also https://github.com/projectcontour/contour/pull/3347#issuecomment-776954782 as it is related

**Comment by youngnick** [comment]
Okay, so our current state on cipher deprecations is:
- cipher configurability is now in place (with #3292 merged and #2880 closed out)
- we will document the current default and available cipher suite.

Open to opinions from operators on the latter item. We can either document this by listing the default and supported cipher suite in a separate document, or ensure that the TLS documentation refers to the source `.go` file where we define those two things. The former is easier to read, the latter is easier to keep up to date.

@moderation and anyone else, thoughts? I think that once we've got agreement on how to do the docs, I'll open a new issue to cover changing the docs, and close this one out.

**Comment by youngnick** [comment]
The cipher suite documentation is available in the docs at https://projectcontour.io/docs/v1.13.0/configuration/#tls-configuration. I'll close this one out as a result.

@moderation, please respond here if you disagree.

**Comment by moderation** [comment]
LGTM. Thanks to @youngnick and the Contour team
